Client/trunk/ParaEngineClient/IO/ZipArchive.cpp

```diff
@@ -562,11 +591,46 @@ int CZipArchive::findFile(const string& sFilename)
 	}
 
 	if (m_bIgnoreCase)
+	{
 		StringHelper::make_lower(entry.zipFileName);
+	}
+
+	
+	entry.RefreshHash(false);
+
+
+	Sort();
 
-	int res = m_FileList.binary_search(SZipFileEntryPtr(&entry));
+	auto it = std::lower_bound(m_FileList.begin(), m_FileList.end(), entry, [](const SZipFileEntryPtr& a, const SZipFileEntry& b)
+	{
+		return a.m_pEntry->hashValue < b.hashValue;
+
+	});
 
-	return res;
+	if (m_bIgnoreCase)
+	{
+		for (; it != m_FileList.end() && it->m_pEntry->hashValue == entry.hashValue; it++)
+		{
+			auto filename = it->m_pEntry->zipFileName;
+			StringHelper::make_lower(filename);

```

Comment:
try to use a char str[512] here instead of a new heap string for each search call.